File: Scala/problem09.scala
def sqr(y: Double): Double = {
  scala.math.pow(y, 2)
}
	
def problem9(x: Int): Option[Int] = {
  val sqrx = sqr(x) // let in
  def fb(a: Double): Double = (sqrx - 2*x*a)/(2*(x-a))
  def fc(a: Double, b: Double): Double = scala.math.sqrt(sqr(a) + sqr(b))
  // use streams with filters and find instead
  val solution = (for {
    a <- (1 to x/3)
    val b: Double = fb(a.toDouble)
    if b.isWhole
    val c: Double = fc(a.toDouble, b.toDouble)
    if c.isWhole
  } yield (a * b * c).toInt ).headOption
  
  solution
}

File: python/problem09.py
# sqrt(a**2 + b**2) + a + b = 1000
# a < b
def sqr(x):
    return x ** 2

def is_int(x):
    return float(x).is_integer()

def problem9(x):
    
    sqrx = sqr(x)
    
    def fb(a):
        return (sqrx - 2*x*a)/(2*(x - a))
    
    def fc(a, b):
        return (sqr(a) + sqr(b)) ** 0.5
    
    for a in range(1, x/3 + 1):
        b = fb(a)
        if is_int(b):
            c = fc(a, b)
            if is_int(c):
                return (a * b * c)

